oop/src/components/DateInput.vue

```diff
@@ -14,27 +14,29 @@
           <option v-for="(month, index) in monthList" :key="index" :value="index">{{month}}</option>
         </select>
 
-        <label for="dayInput">Day:</label>
-        <input id="dayInput"
-              type="number"
-              class="form-control dayInput"
-              placeholder="Day"
-              v-model="day"
-              @blur="onBlurDay($event.target.value)"
-              :disabled='disabled'
-              maxlength="2"
-              autocomplete="off"/>
-
-        <label for="yearInput">Year:</label>
-        <input id="yearInput"
-              type="number"
-              class="form-control yearInput"
-              placeholder="Year"
-              v-model="year"
-              @blur="onBlurYear($event.target.value)"
-              :disabled='disabled'
-              maxlength="4"
-              autocomplete="off"/>
+        <label v-bind:for="'dayInput' + label">Day:</label>
+        <input 
+            :id="'dayInput' + label"
+            class="form-control dayInput"
+            placeholder="DD"
+            v-model="day"
+            @blur="onBlurDay($event.target.value)"
+            :disabled='disabled'
+            maxlength="2"
+            v-on:keypress="isNumber($event)"
+            >
+
+        <label v-bind:for="'yearInput' + label">Year:</label>
+        <input 
+            :id="'yearInput' + label"
+            class="form-control dayInput"
+            placeholder="YYYY"
+            v-model="year"
+            @blur="onBlurYear($event.target.value)"
+            :disabled='disabled'
+            maxlength="4"
+            v-on:keypress="isNumber($event)"
+            >
```

Comment:
Instead of using the `v-on:keypress` and filtering of numeric keys, can you use the `vue-text-mask` package instead? By using the text mask regex, we can filter out digits that way.

See `PhnInput` as an example from the common lib: https://github.com/bcgov/common-lib-vue/blob/main/src/components/PhnInput.vue#L4-L13